### PR TITLE
Algorithm (PRE-MIDPOINT FEEDBACK WORK)

### PR DESCRIPTION
**Pull request by HariniSundaram0**
# Summary
In these series of commits, I updated my algorithm to query from Spotify's web API, and select a random element instead of the previously hard coded version. This feature mainly focuses on implementing the network and parsing requests. 

# Description:
I implemented a basic network request to the Spotify's Web API, using the search endpoint. It currently queries for 50 songs, with a search of 'a'. Then, I parsed it via a series of conditional downcasting statements, and then selects a random element. I originally attempted to map the nested JSON response into a flat model by using the 'decoder' JSON protocol. However since I only wanted a single uri, I found that creating structs and multiple objects were unnecessary. Please let me know if you believe that another method for parsing would be better.

# Test Plan:
In my preliminary testing, I noticed that if you swipe too many times, the Spotify connection will crash. I found a stack overflow article on this, so fixing this edge case will become a secondary task for the upcoming week. (I apologize for the large commit.)

### CODE REVIEW

File: Project-Hotpot/Project-Hotpot/Algorithm/SongAlgorithm.swift [hunk 1]
```diff
@@ -10,37 +10,46 @@ import Parse
 import AFNetworking
 
 class songAlgorithm{
```
Comment (harshsinha) [inline Project-Hotpot/Project-Hotpot/Algorithm/SongAlgorithm.swift]: Class names are always Capital case: SongAlgoritm

File: Project-Hotpot/Project-Hotpot/ViewControllers/HomeViewController.swift [hunk 2]
```diff
@@ -73,16 +72,14 @@ class HomeViewController: UIViewController {
...
                 UIView.animate(withDuration: 0.2, delay: 2.0) {
                     self.resetSong()
                     self.resetCard()
                 }
-                
                 return
-                //move off to right
             }
-            resetCard()
+//            self.resetSong()
```
Comment (harshsinha) [inline Project-Hotpot/Project-Hotpot/ViewControllers/HomeViewController.swift:81]: remove?

File: Project-Hotpot/Project-Hotpot/ViewControllers/HomeViewController.swift [hunk 3]
```diff
@@ -120,20 +116,28 @@ class HomeViewController: UIViewController {
...
+                NSLog(uri ?? "nil uri call")
+                self.api_instance.appRemote.contentAPI?.fetchContentItem(forURI: songURI!, callback: {success, error in
+
+                    if (error != nil) {
+                        NSLog(error?.localizedDescription ?? "error fetching song")
+                    }
+                    else if let success = success as? SPTAppRemoteContentItem
+                    {
+                        //play if no errors
+                        self.api_instance.appRemote.playerAPI?.play(success)                    }
```
Comment (harshsinha) [inline Project-Hotpot/Project-Hotpot/ViewControllers/HomeViewController.swift:133]: Nit: playerAPI.play(success) 
does not read well, maybe consider changing to some meaningful name like 'song' or 'songContent' etc

File: Project-Hotpot/Project-Hotpot/Algorithm/SongAlgorithm.swift [hunk 4]
```diff
@@ -9,9 +9,7 @@ import Foundation
 import Parse
 import AFNetworking
 
-class songAlgorithm{
-    var defaultSong: String = "spotify:album:3RQQmkQEvNCY4prGKE6oc5"
-    
+class songAlgorithm{    
     //TODO: CRASHES WHEN TOO MANY SONG REQUESTS, FOUND STACKOVERFLOW ALREADY
```
Comment (harshsinha) [inline Project-Hotpot/Project-Hotpot/Algorithm/SongAlgorithm.swift:13]: If this is known, consider showing some kind of error screen to the user like "Too many requests, please wait for sometime and try again"
